core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -282,7 +282,9 @@ private[spark] object Utils extends Logging {
         if (dir.exists() || !dir.mkdirs()) {
           dir = null
         } else {
-          if (!chmod700(dir)) {
+          // Restrict file permissions via chmod if available.
+          // For Windows this step is ignored.
+          if (!isWindows && !chmod700(dir)) {
```

Comment:
Actually, the current ordering seems more intuitive, so I'm going to keep the patch in its current form.